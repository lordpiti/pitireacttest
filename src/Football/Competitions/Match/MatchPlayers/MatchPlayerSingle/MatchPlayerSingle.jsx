import React from 'react';

const MatchPlayerSingle = ( props ) => {

  let substitutionIn, substitutionOut = null;

  if (props.player.substitutionIn) {
    substitutionIn = <span>
      <span className="glyphicon glyphicon-arrow-left"></span>
      <span>{props.player.substitutionIn.minute}</span>
    </span>;
  }

  if (props.player.substitutionOut) {
    substitutionOut = <span>
      <span className="glyphicon glyphicon-arrow-right"></span>
      <span>{props.player.substitutionOut.minute}</span>
    </span>;
  }

  return (
    <div className="row match-player-single">
    <div className="col-sm-1">{props.player.dorsal}</div>
    <div className="col-sm-5">
      {props.player.name} {props.player.surname}
    </div>
    <div className="col-sm-1">
        {substitutionIn}
        {substitutionOut}
    </div>
    <div className="col-sm-2">
      {props.player.bookings.map((booking, index) => {
        const cardImageUrl = booking.type==='Amarilla'?'/assets/img/yellow_card-512.png':'/assets/img/red_card-512.png';

        return <div key={index}>
          <img src={cardImageUrl} width="20" height="20" alt=""/>
          <label className="small">{booking.minute}</label>
        </div>
      }
      )}
    </div>
    <div className="col-sm-2">
      {props.player.goals.map((goal, index) =>
        <span key={index}>
          <img src="/assets/img/ball.png" width="20" height="20" alt=""/>
          <label className="small">{goal.minute}</label>
        </span>)
      }
    </div>
  </div>
  )
};

export default MatchPlayerSingle;