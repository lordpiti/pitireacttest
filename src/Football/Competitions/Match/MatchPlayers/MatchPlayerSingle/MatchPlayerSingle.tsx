import React from 'react';

interface PlayerMatchSingle {
  name: string;
  surname: string;
  dorsal: string;
  substitutionIn: Substitution;
  substitutionOut: Substitution;
  bookings: Booking[];
  goals: Goal[];
}

interface Booking {
  minute: number;
  type: string;
}

interface Goal {
  minute: number;
}

interface MatchPlayerSingleProps {
  player: PlayerMatchSingle;
}

interface Substitution {
  minute: number;
}

const MatchPlayerSingle = (props: MatchPlayerSingleProps) => {
  return (
    <div className='row match-player-single'>
      <div className='col-sm-1'>{props.player.dorsal}</div>
      <div className='col-sm-5'>
        {props.player.name} {props.player.surname}
      </div>
      <div className='col-sm-1'>
        {props.player.substitutionIn && (
          <span>
            <span className='glyphicon glyphicon-arrow-left'></span>
            <span>{props.player.substitutionIn.minute}</span>
          </span>
        )}
        {props.player.substitutionOut && (
          <span>
            <span className='glyphicon glyphicon-arrow-right'></span>
            <span>{props.player.substitutionOut.minute}</span>
          </span>
        )}
      </div>
      <div className='col-sm-2'>
        {props.player.bookings.map((booking, index) => {
          const cardImageUrl =
            booking.type === 'Amarilla'
              ? '/assets/img/yellow_card-512.png'
              : '/assets/img/red_card-512.png';

          return (
            <div key={index}>
              <img src={cardImageUrl} width='20' height='20' alt='' />
              <label className='small'>{booking.minute}</label>
            </div>
          );
        })}
      </div>
      <div className='col-sm-2'>
        {props.player.goals.map((goal, index) => (
          <span key={index}>
            <img src='/assets/img/ball.png' width='20' height='20' alt='' />
            <label className='small'>{goal.minute}</label>
          </span>
        ))}
      </div>
    </div>
  );
};

export default MatchPlayerSingle;
