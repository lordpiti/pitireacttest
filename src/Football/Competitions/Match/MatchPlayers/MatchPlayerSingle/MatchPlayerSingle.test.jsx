import React from 'react';
import renderer from 'react-test-renderer';

import MatchPlayerSingle from './MatchPlayerSingle';

describe('<MatchPlayerSingle /> with all data', () => {
    it('renders correctly a player component with the right data', () => {
        const playerData = {
            name: 'test name',
            surname: 'test surname',
            dorsal: 9,
            bookings: [{
                type: 'Amarilla',
                minute: 10
            }],
            goals: [{
                minute: 76
            }],
            substitutionIn: {
                minute: 9
            },
            substitutionOut: {
                minute: 19
            },
        };
      const tree = renderer.create(<MatchPlayerSingle player={playerData} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
})