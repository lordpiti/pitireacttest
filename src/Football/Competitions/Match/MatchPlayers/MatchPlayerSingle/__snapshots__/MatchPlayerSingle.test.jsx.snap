// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchPlayerSingle /> with all data renders correctly a player component with the right data 1`] = `
<div
  className="row match-player-single"
>
  <div
    className="col-sm-1"
  >
    9
  </div>
  <div
    className="col-sm-5"
  >
    test name
     
    test surname
  </div>
  <div
    className="col-sm-1"
  >
    <span>
      <span
        className="glyphicon glyphicon-arrow-left"
      />
      <span>
        9
      </span>
    </span>
    <span>
      <span
        className="glyphicon glyphicon-arrow-right"
      />
      <span>
        19
      </span>
    </span>
  </div>
  <div
    className="col-sm-2"
  >
    <div>
      <img
        alt=""
        height="20"
        src="/assets/img/yellow_card-512.png"
        width="20"
      />
      <label
        className="small"
      >
        10
      </label>
    </div>
  </div>
  <div
    className="col-sm-2"
  >
    <span>
      <img
        alt=""
        height="20"
        src="/assets/img/ball.png"
        width="20"
      />
      <label
        className="small"
      >
        76
      </label>
    </span>
  </div>
</div>
`;
